vllm/engine/arg_utils.py

```diff
@@ -1323,7 +1344,10 @@ def create_engine_config(self,
                                         if self.enable_prompt_adapter else None
 
         decoding_config = DecodingConfig(
-            guided_decoding_backend=self.guided_decoding_backend)
+            guided_decoding_backend=self.guided_decoding_backend,
+            reasoning_backend=self.reasoning_parser
```

Comment:
oh do we have a reasoning_backend on DecodingConfig now?

probably also need to update `vllm/config.py` as well

Reply:
https://github.com/vllm-project/vllm/pull/12955/files#diff-7eaad0b7dee0626bf29d10081b0f0c5e3ea15a4af97e7b182a4e0d35f8346953R2710

Yes, we have the code. However, let me test it locally end-to-end first. It’s been a while since the initial commit, and I haven’t run it after the refactor, so I want to ensure everything works as expected.

Reply:
It should work.